fsw/public_inc/crypto_props.h

```diff
@@ -0,0 +1,138 @@
+==========================================================*/
+void crypto_config_free(crypto_config_list** pp_self);
+
+/*===========================================================================
+Function:           crypto_config_props_add_property      
+Description:        adds a new key & value node to the end of the linkedlist.      
+Inputs:             char* key - some string 
+ *                  char* value- some string          
+Outputs:            1 - success 
+ *                  0 - fail       
+References: 
+Example call:       crypto_config_props_add_property("key_a","some value");
+Note:               since this is a linked list there is no efficient way to guard against
+                    adding the duplicate keys therefore if the key exists just update it using
+                    crypto_config_props_set_property_value_helper else add in a new key,value. 
+==========================================================*/
+int crypto_config_props_add_property(char* key, char* value);
+
+/*===========================================================================
+Function:           crypto_config_get_property_value      
+Description:        if a node exists with that key then the string value is 
+                    returned else "/0" string is returned.      
+Inputs:             char* key - some string key       
+Outputs:            string or "/0"       
+References:
+Example call:       crypto_config_get_property_value("key_a")
+==========================================================*/
+char* crypto_config_get_property_value(char* key); 
+
+/*===========================================================================
+Function:           crypto_config_set_property_value      
+Description:        If a node with the input key exists its value will be
+ *                  overwritten with input parameter char* value; 
+Inputs:             char* key- some string value         
+Outputs:            1 - success 
+ *                  0 - fail       
+References: 
+Example call:      crypto_config_set_property_value("key_a","AAA")
+==========================================================*/
+int crypto_config_set_property_value(char* key, char* value);
+
+/*===========================================================================
+Function:      
+Description:        This function will parse files with the format:
+                    #Comment#
+                    Key=value,
+                    key=value,
+                    OR:
+                    #Comment
+ *                  //Comment 
+                    Key=value,key=value,key=value
+/*Inputs:           char* file_path - path of the file on the filesystem
+Outputs:            1 - success 
+ *                  0 - fail  
+References:
+Example call:      crypto_config_load_properties("/root/Desktop/config_props.txt")
+==========================================================*/
+int crypto_config_load_properties(char* file_path);
+
```

Comment:
This function is OK to have, but we also want a function to load properties passed directly to CryptoLib... The expectation is that apps will pass in properties to this library rather than a path to a property file. This functionality can be added later though...

Reply:
Regarding thw function crypto_config_load_properties(char* file_path) in the file crypto_props.c:
**In order:**

1) Now that you caught that issue, I’m guarding against have less than 2 chars in the function crypto_config_load_properties(char* file_path) lines 281-286.

2) I’m not actually validating against alphanumeric characters, however I’m checking against whether the current parsed line is a “data line” (key=value) vs #comment or empty line that should be ignored. 

**Furthermore, as per specs in the SIS non-alphanumeric characters are allowed for not the key & values r, and here are some examples on non-alphanumeric key values that are mentioned in the SIS (pages 10-11):**

key_management_service_uri=tls12://fully-qualified-domain-name:5696
The key has char "_" which is non alphanumeric & the value has chars ":","//",and "-"

**and here is another example:**

allowed_symmetric_encryption_algorithms=AES-128:AES-256:DESede-168
The key has char "_" which is non-alphanumeric & the value has chars ":", and "-"

Regardless I changed the code (current_line[1]!='/' || current_line[0]!='/') to (current_line[1]!='/' && current_line[0]!='/')  in order wrap up this converstaion, although this will NOT allow keys such as:

/key=some_value which is totally valid, but ugly. 

These new changes will in the new pull request.